cmd/goal/node.go

```diff
@@ -290,7 +290,22 @@ func getStatus(dataDir string) {
 func makeStatusString(stat v1.NodeStatus) string {
 	lastRoundTime := fmt.Sprintf("%.1fs", time.Duration(stat.TimeSinceLastRound).Seconds())
 	catchupTime := fmt.Sprintf("%.1fs", time.Duration(stat.CatchupTime).Seconds())
-	return fmt.Sprintf(infoNodeStatus, stat.LastRound, lastRoundTime, catchupTime, stat.LastVersion, stat.NextVersion, stat.NextVersionRound, stat.NextVersionSupported, stat.HasSyncedSinceStartup)
+	statusString := fmt.Sprintf(
+		infoNodeStatus,
+		stat.LastRound,
+		lastRoundTime,
+		catchupTime,
+		stat.LastVersion,
+		stat.NextVersion,
+		stat.NextVersionRound,
+		stat.NextVersionSupported)
+
+	if stat.LastRound+1 == stat.NextVersionRound &&
```

Comment:
That logic should not be here; to be more specific, the predicate
`stat.LastRound+1 == stat.NextVersionRound && !stat.NextVersionSupported` should be a bool field in the `v1.NodeStatus` struct that indicates whether the node is capable of making a progress or not.
It's the `AlgorandFullNode::Status()` that should fill up this value, and have it propagated all the way upstream.